src/features/core/deviceModeTransformation/dmtHandler.js

```diff
@@ -0,0 +1,190 @@
+/* eslint-disable compat/compat */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-param-reassign */
+/* eslint-disable prefer-promise-reject-errors */
+/* eslint-disable consistent-return */
+import { replacer, removeTrailingSlashes } from '../../../utils/utils';
+import logger from '../../../utils/logUtil';
+import { createPayload } from './util';
+
+class TransformationsHandler {
+  constructor() {
+    this.retryAttempt = 3; // default value for retry
+    this.queue = [];
+    this.isTransformationProcessing = false;
+    this.authToken = null;
+  }
+
+  init(writeKey, dataPlaneUrl, authToken) {
+    this.dataPlaneUrl = removeTrailingSlashes(dataPlaneUrl);
+    this.writeKey = writeKey;
+    this.authToken = authToken || this.authToken;
+    this.start();
+  }
+
+  // Enqueue the events and callbacks
+  enqueue(event, cb) {
+    this.queue.push({
+      event,
+      cb,
+    });
+  }
+
+  /**
+   * A helper function that will take one payload at a time and send it to transformation server
+   * and return the response
+   *
+   */
+  sendEventForTransformation(payload, retryAttempt) {
+    return new Promise((resolve, reject) => {
+      const url = `${this.dataPlaneUrl}/transform`;
+      const headers = {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${btoa(`${this.writeKey}:`)}`,
+      };
+
+      try {
+        const xhr = new XMLHttpRequest();
+        xhr.open('POST', url, true);
+        Object.keys(headers).forEach((k) => xhr.setRequestHeader(k, headers[k]));
+
+        xhr.onreadystatechange = () => {
+          if (xhr.readyState === 4) {
+            try {
+              const { status } = xhr;
+              let { response } = xhr;
+
+              if (status === 200) {
+                if (response && typeof response === 'string') {
+                  response = JSON.parse(response);
```

Comment:
shoudl we try responseText instead?

Reply:
can we also set a custom timeout for the request?

Reply:
I will address it.

Reply:
Done